parser.y

```diff
@@ -8249,47 +8257,133 @@ SelectStmtLimitOpt:
 	}
 |	SelectStmtLimit
 
+SelectStmtOpt:
+	TableOptimizerHints
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.TableHints = $1.([]*ast.TableOptimizerHint)
+		$$ = opt
+	}
+|	DistinctOpt
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		if $1.(bool) {
+			opt.Distinct = true
+		} else {
+			opt.Distinct = false
+			opt.ExplicitAll = true
+		}
+		$$ = opt
+	}
+|	Priority
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.Priority = $1.(mysql.PriorityEnum)
+		$$ = opt
+	}
+|	SelectStmtSQLSmallResult
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.SQLSmallResult = $1.(bool)
+		$$ = opt
+	}
+|	SelectStmtSQLBigResult
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.SQLBigResult = $1.(bool)
+		$$ = opt
+	}
+|	SelectStmtSQLBufferResult
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.SQLBufferResult = $1.(bool)
+		$$ = opt
+	}
+|	SelectStmtSQLCache
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = $1.(bool)
+		$$ = opt
+	}
+|	SelectStmtCalcFoundRows
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.CalcFoundRows = $1.(bool)
+		$$ = opt
+	}
+|	SelectStmtStraightJoin
+	{
+		opt := &ast.SelectStmtOpts{}
+		opt.SQLCache = true
+		opt.StraightJoin = $1.(bool)
+		$$ = opt
+	}
+
 SelectStmtOpts:
-	TableOptimizerHints DefaultFalseDistinctOpt PriorityOpt SelectStmtSQLSmallResult SelectStmtSQLBigResult SelectStmtSQLBufferResult SelectStmtSQLCache SelectStmtCalcFoundRows SelectStmtStraightJoin
+	%prec empty
```

Comment:
If we change to `SelectStmtOptsList`, the order of hints & selection options can be arbitrary. Is it expected?

Reply:
Yeah. I test it in MySQL.